library/core/src/error.rs

```diff
@@ -412,6 +412,60 @@ impl dyn Error {
     }
 }
 
+/// Request a value from the `Error`.
+///
+/// # Examples
+///
+/// Get a string value from an error.
+///
+/// ```rust
+/// # #![feature(error_generic_member_access)]
+/// use std::error::Error;
+///
+/// fn get_string(err: &impl Error) -> String {
+///     (err as &dyn Error).request_value::<String>().unwrap()
+/// }
+/// ```
```

Comment:
This doc comment should be moved into the `impl dyn Error` version of this function that is actually `pub`.